libcontainer/rootfs_linux.go

```diff
@@ -35,6 +35,7 @@ type mountConfig struct {
 	rootlessCgroups bool
 	cgroupns        bool
 	fd              *int
+	idmapFd         *int
```

Comment:
I'm unsure if we want to have fd and idmapFd or just have one fd field here and when the mount is idmapped it is the idmap fd, when it is a mountFd then the fd field is that and when none is needed is nil.

@kolyshkin what do you think?

Reply:
First of all, there should not be any fd here, since `mountConfig` is not about a particular mount. This was fixed by commit a60933bb245657e3e1f3 (and you need to change your code).

I think that we can and should reuse `mountEntry.srcFD` field.

Reply:
This was already done by Francis. It seems reusing srcFD is not straight forward as that is a string and we need an int for the idmap case.

Even if we use a string, we need to know when it is a srcFD, so we add /proc/self/fd/ and in the idmap case we don't need to do that.

There might be a way to do it, but it seems error-prone. If you disagree @kolyshkin , please re-open :)